src/main/scala/com/johnsnowlabs/nlp/annotators/sbd/SentenceDetectorParams.scala

```diff
@@ -38,4 +39,31 @@ trait SentenceDetectorParams extends Params {
 
   def getMaxLength: Int = $(maxLength)
 
+  def truncateSentence(sentence: String, maxLength: Int): Array[String] = {
+    var currentLength = 0
+    val allSentences = ArrayBuffer.empty[String]
+    val currentSentence = ArrayBuffer.empty[String]
+
+    def addWordToSentence(word: String): Unit = {
+      /** Adds +1 because of the space joining words */
+      currentLength += word.length + 1
```

Comment:
Is there a reason to set the max length by number of characters instead of number of tokens?

Reply:
Limiting the length of sentence to a number of tokens is not a good idea. If you fail to tokenize correctly like tokenizing the entire URL as one token, if you have multiple URLs, you will end up having 10 tokens in a sentence each around 50 characters (500 characters). Limiting the length by the number of tokens defeats the purpose of avoiding large sentences. 

Really hard to set a correct size to limit if it was the number of tokens. (limit by number of characters is the best practice same as in CoreNLP)

Reply:
That's an interesting point about tokenization errors that result from missing spaces between words, or in long URLS.  However, this addition that requires splitting on whitespace will prevent this solution from solving the problem you just outlined @maziyarpanahi .  This feature may be worth thinking about in more detail.

I believe this feature was created to address another issue around a dl model having a max length setting that was throwing errors when embedded sentences were too long (too many tokens), an issue which this pr would theoretically solve.

I think that is why the maxLength setting is turned off by default.

Reply:
Annotators should be capable of handling their own limits if needed. This change is towards that a little bit. This sentence limit might help universal problems or limitations or very specific use cases, but ideally, annotator specific limitations should be addressed locally.

Agreed on the points pointed out above!